crates/sui-types/src/messages_checkpoint.rs

```diff
@@ -387,21 +379,40 @@ impl CertifiedCheckpointSummary {
     }
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct CheckpointProposalContents {
+    // TODO: Currently we are not really using the effects digests, but in the future we may be
+    // able to use it to optimize the sync process.
+    pub transactions: BTreeSet<ExecutionDigests>,
```

Comment:
proposal contents are _not_ assumed to be causally ordered, correct? Can you document this?